craft_parts/permissions.py

```diff
@@ -43,40 +43,38 @@ class Permissions(BaseModel):
     """
 
     path: str = "*"
-    owner: Optional[int] = None
-    group: Optional[int] = None
-    mode: Optional[str] = None
+    owner: int | None = None
+    group: int | None = None
+    mode: str | None = None
 
-    # pylint: disable=no-self-argument
     @root_validator(pre=True)
-    def validate_root(cls, values: Dict[Any, Any]) -> Dict[Any, Any]:
+    def validate_root(cls, values: dict[Any, Any]) -> dict[Any, Any]:
         """Validate that "owner" and "group" are correctly specified."""
         has_owner = "owner" in values
         has_group = "group" in values
 
-        assert (
-            has_group == has_owner
-        ), 'If either "owner" or "group" is defined, both must be'
+        if has_group != has_owner:
+            raise ValueError(
+                'If either "owner" or "group" is defined, both must be the same.'
```

Comment:
```suggestion
                'If either "owner" or "group" is defined, both must be.'
```

It's not that "owner" and "group" have to have the same value, it's that if the user defines one of them then they must also define the other.

Reply:
OK